### PR TITLE
Implement data manager testing and update bowtie2 data manager

### PR DESCRIPTION
**Pull request by mvdbeek**
FOR CONTRIBUTOR:
* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by mvdbeek** [comment]
Alright, so missing test files will cause a test failure: https://travis-ci.org/mvdbeek/tools-iuc/jobs/530294667
That is exactly what I intended ... 😆

**Comment by mvdbeek** [comment]
Hooray, it's passing! https://travis-ci.org/mvdbeek/tools-iuc/jobs/530301638

File: data_managers/data_manager_bowtie2_index_builder/data_manager/bowtie2_index_builder.py [hunk 1]
```diff
@@ -31,6 +31,9 @@ def build_bowtie2_index( data_manager_dict, fasta_filename, params, target_direc
     sym_linked_fasta_filename = os.path.join( target_directory, fasta_base_name )
     os.symlink( fasta_filename, sym_linked_fasta_filename )
     args = [ 'bowtie2-build', sym_linked_fasta_filename, sequence_id ]
+    threads = os.environ.get('GALAXY_SLOTS')
```
Comment (bgruening) [inline data_managers/data_manager_bowtie2_index_builder/data_manager/bowtie2_index_builder.py]:
```suggestion
    threads = os.environ.get('GALAXY_SLOTS', None)
```

Comment (mvdbeek) [reply data_managers/data_manager_bowtie2_index_builder/data_manager/bowtie2_index_builder.py]: That is the same thing

**Comment by bernt-matthias** [comment]
Hi @mvdbeek. For curiosity: how do I run a planemo test for a data manager. I always get errors 

```
Traceback (most recent call last):
  File "/tmp/tmpwjkom4/galaxy-dev/lib/galaxy/jobs/runners/local.py", line 152, in queue_job
    self._finish_or_resubmit_job(job_state, stdout, stderr, exit_code)
  File "/tmp/tmpwjkom4/galaxy-dev/lib/galaxy/jobs/runners/__init__.py", line 459, in _finish_or_resubmit_job
    job_state.job_wrapper.finish(stdout, stderr, exit_code, check_output_detected_state=check_output_detected_state)
  File "/tmp/tmpwjkom4/galaxy-dev/lib/galaxy/jobs/__init__.py", line 1464, in finish
    self.tool.exec_after_process(self.app, inp_data, out_data, param_dict, job=job)
  File "/tmp/tmpwjkom4/galaxy-dev/lib/galaxy/tools/__init__.py", line 2317, in exec_after_process
    assert data_manager is not None, "Invalid data manager (%s) requested. It may have been removed before the job completed." % (data_manager_id)
AssertionError: Invalid data manager (bowtie2_index_builder_data_manager) requested. It may have been removed before the job completed.
FAIL
```

When I run `planemo t` it the root of the data manager...

**Comment by mvdbeek** [comment]
You have to point planemo to the data manager tool in data_manager/tool.xml

**Comment by bernt-matthias** [comment]
Cool. Thanks a lot.

**Comment by bernt-matthias** [comment]
One more question: Is it expected that the test changes the loc file in the `test-data` directory? Each run seems to add a line. Also a  comparison with the loc files in `test-data` seems not to happen. Is this expected?

**Comment by mvdbeek** [comment]
Yes to both. We just check for the expected data_manager.json file. The rest is Galaxy territory. We should copy the test data though before running the test, so that you keep the loc files in the test-data dir clean.

**Pull request merged**